### PR TITLE
Karel/cmc api

### PR DESCRIPTION
**Pull request by KaiserKarel**
Adds an API server for coinmarketcap to consume.

### CODE REVIEW

**Review by andyjsbell — APPROVED** [review]

File: code/services/cmc-api/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	let addr = cli.address;
+	tracing::info!("listening on {}", addr);
+	axum::Server::bind(&addr).serve(app.into_make_service()).await.unwrap();
+}
+
+async fn healthcheck() -> &'static str {
+	"Hello, World!"
+}
+
+const BILLION: u128 = 1_000_000_000_000;
```
Comment (blasrodri) [inline code/services/cmc-api/src/main.rs:28]: Isn't a billion 10^9?

Comment (KaiserKarel) [reply code/services/cmc-api/src/main.rs:28]:
```suggestion
const BILLION: u128 = 1_000_000_000;
```

File: code/services/cmc-api/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+anyhow = "1.0.66"
+async-trait = "0.1.58"
+axum = "0.6.0"
+axum-macros = "0.3.0"
+clap = { version = "4.0.27", features = ["derive"] }
+serde = { version = "1.0.148", features = ["derive"] }
+tokio = { version = "1.22.0", features = ["full"] }
+tracing = "0.1.37"
+tracing-subscriber = "0.3.16"
+url = { version = "2.3.1", features = ["serde"] }
```
[the remark this answers is not in the document]
Comment (KaiserKarel) [reply code/services/cmc-api/Cargo.toml:18]: 👍 
left over from prototyping. We should re-enable udeps again.

**Review by benluelo — APPROVED** [review]

**Review by Martin-Kiss — APPROVED** [review]

**Review by easteregg — APPROVED** [review]

**Review by obsessed-cake — APPROVED** [review]

File: code/services/cmc-api/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+const BILLION: u128 = 1_000_000_000;
+
+#[axum_macros::debug_handler]
+async fn total_supply() -> Result<String, StatusCode> {
+	Ok((BILLION * 10).to_string())
+}
+
+#[axum_macros::debug_handler]
+async fn circulating_supply() -> Result<String, StatusCode> {
+	Ok((BILLION + BILLION / 2).to_string())
```
Comment (vimukthi-git) [inline code/services/cmc-api/src/main.rs:37]: How do we arrive at this number?

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: code/services/cmc-api/src/main.rs (hunk 3 above)
Comment (KaiserKarel) [reply code/services/cmc-api/src/main.rs:37]: Currently the crowdloaners :) Will update this as we start vestings for investors/make it dynamic.

**Pull request merged**